pkg/helm/informers.go

```diff
@@ -0,0 +1,178 @@
+	}
+	nsMon := monitorMap[targetNamespace]
+	monitorMux.Unlock()
+
+	go runAppStateMonitor(nsMon)
+}
+
+func resumeHelmStatusInformers(appName string) {
+	helmApp := GetHelmApp(appName)
+	app := &apptypes.App{
+		ID:                helmApp.GetID(),
+		Slug:              helmApp.GetSlug(),
+		Name:              helmApp.Release.Name,
+		IsAirgap:          helmApp.GetIsAirgap(),
+		CurrentSequence:   helmApp.GetCurrentSequence(),
+		UpdatedAt:         &helmApp.CreationTimestamp,
+		CreatedAt:         helmApp.CreationTimestamp,
+		LastUpdateCheckAt: &helmApp.CreationTimestamp,
+		IsConfigurable:    helmApp.IsConfigurable,
+	}
+
+	kotsKinds, err := GetKotsKindsFromHelmApp(helmApp)
+	if err != nil {
+		logger.Errorf("failed to get kots kinds from helm app: %v\n", err)
+		return
+	}
+
+	settings := registrytypes.RegistrySettings{
+		Namespace: util.PodNamespace,
+	}
+	builder, err := render.NewBuilder(&kotsKinds, settings, app.Slug, helmApp.GetCurrentSequence(), app.IsAirgap, util.PodNamespace)
+	if err != nil {
+		logger.Errorf("failed to create new builder: %v\n", err)
+		return
+	}
+
+	// apply the informers for this app since they havent been applied yet
+	if err := applyStatusInformers(app, helmApp.GetCurrentSequence(), &kotsKinds, builder, helmApp.Namespace); err != nil {
+		logger.Errorf("failed to apply status informers: %v\n", err)
+		return
+	}
+}
+
+func applyStatusInformers(a *apptypes.App, sequence int64, kotsKinds *kotsutil.KotsKinds, builder *template.Builder, namespace string) error {
+	renderedInformers := []appstatetypes.StatusInformerString{}
+
+	// render status informers
+	for _, informer := range kotsKinds.KotsApplication.Spec.StatusInformers {
+		renderedInformer, err := builder.String(informer)
+		if err != nil {
+			logger.Errorf("failed to render status informer: %v\n", err)
+			continue
+		}
+		if renderedInformer == "" {
+			continue
+		}
+		renderedInformers = append(renderedInformers, appstatetypes.StatusInformerString(renderedInformer))
+	}
+
+	if identitydeploy.IsEnabled(kotsKinds.Identity, kotsKinds.IdentityConfig) {
+		renderedInformers = append(renderedInformers, appstatetypes.StatusInformerString(fmt.Sprintf("deployment/%s", identitytypes.DeploymentName(a.Slug))))
+	}
+
+	if len(renderedInformers) > 0 {
+		applyAppInformers(a.ID, sequence, renderedInformers, namespace)
+	} else {
+		// no informers, set state to ready
+		defaultReadyState := appstatetypes.ResourceStates{
+			{
+				Kind:      "EMPTY",
+				Name:      "EMPTY",
+				Namespace: "EMPTY",
+				State:     appstatetypes.StateReady,
+			},
+		}
+
+		release := GetHelmApp(a.ID)
+		appStatus := appstatetypes.AppStatus{
+			AppID:          a.ID,
+			ResourceStates: defaultReadyState,
+			Sequence:       release.GetCurrentSequence(),
+			UpdatedAt:      time.Now(),
+		}
+		release.Status = appStatus
+		release.Status.State = appstatetypes.GetState(defaultReadyState)
+		AddHelmApp(release.Release.Name, release)
```

Comment:
Let's not call this here. Modifying the app is sufficient because it's a pointer.